src/client.ml

```diff
@@ -1,10 +1,30 @@
 type t =
-  { api_key : string
-  ; gen_url : string -> string
+  { mutable api_key : string
+  ; mutable url : string
   ; c : Curl.t
+  ; mutable model : string
```

Comment:
I think it's enough to just add `model` field, so why don't you omit `mutable` attributes?